staging/src/k8s.io/csi-translation-lib/plugins/in_tree_volume.go

```diff
@@ -76,38 +77,85 @@ const (
 	zonesKey = "zones"
 )
 
-// replaceTopology overwrites an existing topology key by a new one.
+// replaceTopology overwrites an existing key by a new one.
+// If there are any newKey already exist in a term, we will combine the
+// replaced key Values with the existing ones
 func replaceTopology(pv *v1.PersistentVolume, oldKey, newKey string) error {
+	// Make sure the necessary fields exist
+	if pv == nil || pv.Spec.NodeAffinity == nil || pv.Spec.NodeAffinity.Required == nil ||
+		pv.Spec.NodeAffinity.Required.NodeSelectorTerms == nil || len(pv.Spec.NodeAffinity.Required.NodeSelectorTerms) == 0 {
+		return nil
+	}
 	for i := range pv.Spec.NodeAffinity.Required.NodeSelectorTerms {
+		newKeyValsMap := make(map[string]bool)
+		var requirementsToRemove []int
 		for j, r := range pv.Spec.NodeAffinity.Required.NodeSelectorTerms[i].MatchExpressions {
 			if r.Key == oldKey {
 				pv.Spec.NodeAffinity.Required.NodeSelectorTerms[i].MatchExpressions[j].Key = newKey
 			}
+			if r.Key == newKey || r.Key == oldKey {
```

Comment:
Does this approach work for deduplication: create a new empty list for `MatchExpressions`, and only append values that don't satisfy this if statement

Reply:
Sorry I might not get the case you mentioned here. Can you give a small example? The logic here is that for each term, we iterate the MatchExpression and see if there is a newKey/oldKey exist, if it does, we add it to the `requirementsToRemove` candidate. And after iteration, we check the size of `requirementsToRemove`. It it is greater than 1. It means we have duplication.

If we have a MatchExpressions that has no newKey/oldKey, then it will just skip this term and keep working on next term.

Reply:
Since we are not remove duplication here. I am just going to keep the original logic which is to replace the old key with new key.